### ISSUE TITLE
glide init adds sub packages of top level package to imports

### ISSUE THREAD

**Issue by james-lawrence** [Masterminds/glide#259]
```bash
glide init
```
```yml
package: bitbucket.org/soandso/package
import:
- package: bitbucket.org/soandso/package
  subpackages:
   - A
   - B
   - C
```
then it complains about these sub packages not being in vendor....
it also attempts to fetch the top level package because its in the import set.

is there a reason for this? I thought it would ignore any packages root at the top level package.

**Comment by jrsanz1** [comment]
I'm also seeing this but only on windows version: 0.9.0-rc1
On osx using version : 0.9.0-rc1-4-gd0a4453 the result is empty import.

`package: bitbucket.org/soandso/package
import: []`

**Comment by mattfarina** [comment]
Is the top level package, `bitbucket.org/soandso/package`, inside the `GOPATH` properly? Glide has a step that should detect the top level package being referenced and remove it if present.

What OS, version, and can you share the output of `go env`?

**Comment by jrsanz1** [comment]
I'm able to replicate this on Windows 10 64-bit

```
glide -v
glide version 0.9.0-rc1
```
```
go env
set GOARCH=amd64
set GOBIN=
set GOEXE=.exe
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOOS=windows
set GOPATH=C:\dev\go\
set GORACE=
set GOROOT=C:\go
set GOTOOLDIR=C:\go\pkg\tool\windows_amd64
set GO15VENDOREXPERIMENT=1
set CC=gcc
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0
set CXX=g++
set CGO_ENABLED=1
```

glide.exe is located in $GOROOT and $GOPATH contains the expected 'src' directory

**Comment by mattfarina** [comment]
@jrsanz1 I'm unable to replicate this under Windows. Can you provide some more detail?

Is you codebase somewhere in the `GOPATH\src` and you're operating on it from there rather than a symlink? I ask because I've seen what you describe when the codebase is outside the `GOPATH`.

**Comment by jrsanz1** [comment]
Ok I have discovered why this is happening. Just to clarify my codebase is under GOPATH so all of that is normal. I've been using Visual Studio Code for development, and in has this option to right click on the working folder and select 'Open in command prompt' from there I've been executing my go commands and the 'glide create' which yields the invalid yaml file. If I start up the command prompt normally, navigate to my working folder and run 'glide create' the yaml file with the empty import gets created. Sorry for the confusion this is obviously an issue with my setup and how I'm running VSCode. Thanks.

**Comment by mattfarina** [comment]
@jrsanz1 Thanks for sharing. I had no idea that could happen. It's useful for me to know this for anyone else who comes along with the same problem.

**Comment by james-lawrence** [comment]
going to investigate my env a little bit more. sorry, will get back to you this weekend.

**Comment by james-lawrence** [comment]
seems like something was funky with my environment as well when I ran into this, it is no longer an issue for me. sorry I couldn't provide more details on the exact start that caused it.

**Comment by mattfarina** [comment]
This was fixed in #278.
